File: Web/xdn-voting/daemons/daemonStatus.go
package daemons

import (
	"encoding/json"
	"io"
	"strconv"
	"strings"
	"xdn-voting/coind"
	"xdn-voting/models"
	"xdn-voting/utils"
)

var daemonStatus models.DaemonStatus

func DaemonStatus() {
	blkReqXDN, errNet := utils.GETAny("https://xdn-explorer.com/api/getblockcount")
	if errNet != nil {
		utils.WrapErrorLog(errNet.ErrorMessage() + " " + strconv.Itoa(errNet.StatusCode()))
		return
	}

	bodyXDN, _ := io.ReadAll(blkReqXDN.Body)
	blockhashXDN, _ := strconv.Atoi(string(bodyXDN))
	defer func(Body io.ReadCloser) {
		err := Body.Close()
		if err != nil {
			utils.WrapErrorLog(err.Error())
		}
	}(blkReqXDN.Body)

	daemonStatus.BlockCount = blockhashXDN

	blk, errBlock := coind.WrapDaemon(utils.DaemonWallet, 1, "getblockcount")
	if errBlock != nil {
		utils.WrapErrorLog(errBlock.Error())
		return
	}
	blockCountWallet, errBlock := strconv.Atoi(strings.Trim(string(blk), "\""))
	if errBlock != nil {
		utils.WrapErrorLog(errBlock.Error())
		return
	}
	blkStake, errBlock := coind.WrapDaemon(utils.DaemonStakeWallet, 1, "getblockcount")
	if errBlock != nil {
		utils.WrapErrorLog(errBlock.Error())
		return
	}
	blockCountStakeWallet, errBlock := strconv.Atoi(strings.Trim(string(blkStake), "\""))
	if errBlock != nil {
		utils.WrapErrorLog(errBlock.Error())
		return
	}

	if blockCountWallet == blockhashXDN {
		daemonStatus.Block = true
	} else {
		daemonStatus.Block = false
	}

	if blockCountStakeWallet == blockhashXDN {
		daemonStatus.BlockStake = true
	} else {
		daemonStatus.BlockStake = false
	}
	stake, errBlock := coind.WrapDaemon(utils.DaemonStakeWallet, 1, "getstakinginfo")
	if errBlock != nil {
		utils.WrapErrorLog(errBlock.Error())
		return
	}
	var stakeInfo models.StakingInfo
	_ = json.Unmarshal(stake, &stakeInfo)

	daemonStatus.WalletStake = stakeInfo.Staking

	summary, errNet := utils.GETAny("https://xdn-explorer.com/ext/summary")
	if errNet != nil {
		utils.WrapErrorLog(errNet.ErrorMessage() + " " + strconv.Itoa(errNet.StatusCode()))
		return
	}

	bodySummary, _ := io.ReadAll(summary.Body)
	defer func(Body io.ReadCloser) {
		err := Body.Close()
		if err != nil {
			utils.WrapErrorLog(err.Error())
		}
	}(blkReqXDN.Body)

	var sum models.Summary
	err := json.Unmarshal(bodySummary, &sum)
	if err != nil {
		utils.WrapErrorLog(err.Error())
		return
	}
	daemonStatus.MasternodeCount = sum.Data[0].Masternodecount
	daemonStatus.Difficulty = sum.Data[0].Difficulty
	daemonStatus.HashRate = sum.Data[0].Hashrate
	daemonStatus.CoinSupply = sum.Data[0].Supply

	info, errBlock := coind.WrapDaemon(utils.DaemonWallet, 1, "getinfo")
	if errBlock != nil {
		utils.WrapErrorLog(errBlock.Error())
		return
	}

	var inf models.GetInfo

	err = json.Unmarshal(info, &inf)
	daemonStatus.Version = inf.Version
}

func GetDaemonStatus() models.DaemonStatus {
	return daemonStatus
}

File: Mobile/xdn_new/lib/support/daemon_status.dart
/// block : true
/// blockStake : true
/// stakingActive : false
library;

class DaemonStatus {
  DaemonStatus({
      bool? block, 
      bool? blockStake, 
      bool? stakingActive,
  int? blockCount,
  int? masternodeCount, double? coinSupply, String? hashrate, double? difficulty, String? version}){
    _block = block;
    _blockStake = blockStake;
    _stakingActive = stakingActive;
    _blockCount = blockCount;
    _masternodeCount = masternodeCount;
    _coinSupply = coinSupply;
    _hashrate = hashrate;
    _difficulty = difficulty;
    _version = version;
}

  DaemonStatus.fromJson(dynamic json) {
    _block = json['block'];
    _blockStake = json['blockStake'];
    _stakingActive = json['stakingActive'];
    _blockCount = json['blockCount'];
    _masternodeCount = json['masternodeCount'];
    _difficulty = json['difficulty'];
    _hashrate = json['hashRate'];
    _coinSupply = json['coinSupply'];
    _version = json['version'];

  }
  bool? _block;
  bool? _blockStake;
  bool? _stakingActive;
  int? _blockCount;
  int? _masternodeCount;
  double? _difficulty;
  String? _hashrate;
  double? _coinSupply;
  String? _version;
DaemonStatus copyWith({  bool? block,
  bool? blockStake,
  bool? stakingActive,
  int? blockCount,
  int? masternodeCount,
  double? difficulty,
  String? hashrate,
  double? coinSupply,
  String? version,
}) => DaemonStatus(  block: block ?? _block,
  blockStake: blockStake ?? _blockStake,
  stakingActive: stakingActive ?? _stakingActive,
  blockCount: blockCount ?? _blockCount,
  masternodeCount: masternodeCount ?? _masternodeCount,
  difficulty: difficulty ?? _difficulty,
  hashrate: hashrate ?? _hashrate,
  coinSupply: coinSupply ?? _coinSupply,
  version: version ?? _version,
);
  bool? get block => _block;
  bool? get blockStake => _blockStake;
  bool? get stakingActive => _stakingActive;
  int? get blockCount => _blockCount;
  int? get masternodeCount => _masternodeCount;
  double? get difficulty => _difficulty;
  String? get hashrate => _hashrate;
  double? get coinSupply => _coinSupply;
  String? get version => _version;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['block'] = _block;
    map['blockStake'] = _blockStake;
    map['stakingActive'] = _stakingActive;
    map['blockCount'] = _blockCount;
    map['masternodeCount'] = _masternodeCount;
    map['difficulty'] = _difficulty;
    map['hashrate'] = _hashrate;
    map['coinSupply'] = _coinSupply;
    map['version'] = _version;
    return map;
  }

}